plugins/optimization-detective/storage/class-od-storage-lock.php

```diff
@@ -20,6 +20,44 @@
  */
 final class OD_Storage_Lock {
 
+	/**
+	 * Capability for being able to store a URL Metric now.
+	 *
+	 * @since n.e.x.t
+	 * @access private
+	 * @var string
+	 */
+	const STORE_URL_METRIC_NOW_CAPABILITY = 'od_store_url_metric_now';
+
+	/**
+	 * Adds hooks.
+	 *
+	 * @since n.e.x.t
+	 * @access private
+	 */
+	public static function add_hooks(): void {
+		add_filter( 'user_has_cap', array( __CLASS__, 'filter_user_has_cap' ) );
+	}
+
+	/**
+	 * Filters `user_has_cap` to grant the `od_store_url_metric_now` capability to users who can `manage_options` by default.
+	 *
+	 * @since n.e.x.t
+	 * @access private
+	 *
+	 * @param array<string, bool>|mixed $allcaps Capability names mapped to boolean values for whether the user has that capability.
+	 * @return array<string, bool> Capability names mapped to boolean values for whether the user has that capability.
+	 */
+	public static function filter_user_has_cap( $allcaps ): array {
+		if ( ! is_array( $allcaps ) ) {
+			$allcaps = array();
+		}
```

Comment:
Fair point to include this, although technically someone changing this to not be an array would be `doing_it_wrong`.

It's fine to keep this, but if we already cater for it, I wonder whether we should instead just return the same value in such a scenario. It's basically broken, and we know the condition below won't be satisfied anyway.

If some code does something unexpected, I'd rather just ignore it and pass through, than tamper with what they did wrong and possibly cause _more_ unexpected issues. :)

Reply:
We do this in many other places actually. It's all for the sake of type safety, as otherwise a fatal error would occur since the function returns an `array`. This scenario happens when another plugin incorrectly returns the wrong type on a filter callback.

Reply:
I meant it mostly as: If someone else returns something other than an array, we could also return that same thing. The documented filter input type is basically always the same as the documented filter output type, so from that perspective we could also change the return type here to `array<string, bool>|mixed` and pass through any value that's not `array<string, bool>`.

Reply:
Yeah, although that would require removing the PHP type hint from the function. All of the other filter callbacks in the repo have taken this approach, so if we want to pass through the value unchanged, then all the other callbacks should be updated. 

In any case, this is a safeguard for an edge condition for the sake of type safety.